sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -75,31 +112,46 @@ export interface MessageHandlerOptions {
     maxMessageAutoRenewLockDurationInSeconds?: number;
 }
 
-export { MessagingError }
-
 // @public
-export interface OnError {
-    (error: MessagingError | Error): void;
+export interface MessageHandlers<ContextT> {
+    processError(err: Error): Promise<void>;
+    processMessage(message: ReceivedMessage, context: ContextT): Promise<void>;
 }
 
 // @public
-export interface OnMessage {
-    (message: ServiceBusMessage): Promise<void>;
-}
+export type MessageIterator<ContextT> = AsyncIterable<MessageAndContext<ContextT>>;
+
+export { MessagingError }
 
-// Warning: (ae-forgotten-export) The symbol "Client" needs to be exported by the entry point index.d.ts
-//
 // @public
-export class QueueClient implements Client {
+export interface NonSessionReceiver<LockModeT extends "peekLock" | "receiveAndDelete"> {
     close(): Promise<void>;
-    createReceiver(receiveMode: ReceiveMode): Receiver;
-    createReceiver(receiveMode: ReceiveMode, sessionOptions: SessionReceiverOptions): SessionReceiver;
-    createSender(): Sender;
-    readonly entityPath: string;
-    static getDeadLetterQueuePath(queueName: string): string;
-    readonly id: string;
-    peek(maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-    peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
+    diagnostics: {
+        peek(maxMessageCount?: number): Promise<ReceivedMessage[]>;
+        peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessage[]>;
+    };
+    // (undocumented)
+    entityPath: string;
+    // (undocumented)
+    entityType: "queue" | "subscription";
```

Comment:
How do we envision this being used?

Reply:
Ha, I have a comment in the code that says the exact same thing. A few fields were added as we made the tests work. I'd like to dump it ASAP. 

I'll look at this right after I get all the tests working with the whole parent object idea (soon!)

Reply:
@HarshaNalluru has removed it. :)